### PR TITLE
release 0.4.0

### LINKED ISSUE #14: Fetching data for offline use
The new README shows a method to fetch all data prior to running the first notebook. The last line,

```bash
mdshare fetch *
```

however, cannot (yet) work. We could implement
1. an mdshare-included command line tool to fetch the entire collection or
2. include a script in the tutorial repo to handle the fetching.

Any preferences, @marscher?

### LINKED ISSUE #77: Regular problems with MDShare timeouts
I was trying to run through the notebooks again this morning from a fresh repo clone. I am having problems with md-share timeouts -- I have tried repeatedly the last 30 mins. I mentioned a similar problem some weeks back to @cwehmeyer when I first started looking at the notebooks. Are we confident that this solution will be robust in the wild? 

```TimeoutError: [Errno 60] Operation timed out```

### PR DESCRIPTION
**Pull request by cwehmeyer**
- mdshare now relies on a local catalogue: searching and calling `fetch()` on already downloaded files does not require to be online anymore
- @marscher added progress bars
- `fetch()` can download and automatically extract .tar.gz containers

fixes markovmodel/pyemma_tutorials#14
fixes markovmodel/pyemma_tutorials#77

### CODE REVIEW

File: mdshare/repository.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    def __str__(self):
+        string = ''
+        for key in sorted(self.keys()):
+            size, unit = format_size(self[key]['size']).split(' ')
+            string += '{:50s}   {:6.1f} {}\n'.format(
+                key, float(size), unit)
+        return string.rstrip('\n')
+
+
+class Repository:
```
Comment (marscher) [inline mdshare/repository.py]: derive from object

**Comment by marscher** [comment]
Very nice work! Do you mind adding some info on how to maintain the catalogue? I see there is a template to categorize files according to file patterns, can one change this template easily - or how does it work?

**Review by marscher — APPROVED** [review]

**Comment by thempel** [comment]
Looks great, thanks! 
I checked if tutorial notebooks 0 and 8 still work (w/ and w/o data) and that's the case. The only thing that didn't work as expected was the version number (`mdshare.__version__` returns 0.3.2), but I'm not sure if that is just because it isn't released yet. If so, please ignore.

**Review by thempel — APPROVED** [review]

**Comment by cwehmeyer** [comment]
@marscher and @thempel: thanks a lot for the review!

**Pull request merged**